project/PekkoDisciplinePlugin.scala

```diff
@@ -101,16 +102,18 @@ object PekkoDisciplinePlugin extends AutoPlugin {
   lazy val docs =
     Seq(
       Compile / scalacOptions -= defaultScalaOptions.value,
-      Compile / scalacOptions ++= (
-        if (scalaVersion.value.startsWith("3.")) Nil
-        else Seq("-Wconf:cat=unused:s,cat=deprecation:s,cat=unchecked:s,any:e")
-      ),
+      Compile / scalacOptions ++= (CrossVersion.partialVersion(scalaVersion.value).get match {
+        case (3, _)  => Seq("-Wconf:cat=unused:s,cat=deprecation,cat=other-shadowing:s,any:e")
+        case (2, 13) => Seq("-Wconf:any:e,cat=unused:s,cat=deprecation:s,cat=unchecked:s")
+        case (2, 12) => Seq("-Wconf:cat=unused:s,cat=deprecation:s,cat=unchecked:s,any:e")
+      }),
       Test / scalacOptions --= Seq("-Xlint", "-unchecked", "-deprecation"),
       Test / scalacOptions -= defaultScalaOptions.value,
-      Test / scalacOptions ++= (
-        if (scalaVersion.value.startsWith("3.")) Nil
-        else Seq("-Wconf:cat=unused:s,cat=deprecation:s,cat=unchecked:s,any:e")
-      ),
+      Test / scalacOptions ++= (CrossVersion.partialVersion(scalaVersion.value).get match {
+        case (3, _)  => Seq("-Wconf:cat=unused:s,cat=deprecation,cat=other-shadowing:s,any:e")
+        case (2, 13) => Seq("-Wconf:any:e,cat=unused:s,cat=deprecation:s,cat=unchecked:s")
+        case (2, 12) => Seq("-Wconf:cat=unused:s,cat=deprecation:s,cat=unchecked:s,any:e")
+      }),
```

Comment:
painful to tweak this, hope @som-snytt can back port this to 2.12.x too

Reply:
TBH I'm OK with keeping it around until we drop support for 2.12